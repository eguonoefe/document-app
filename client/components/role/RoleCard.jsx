import React from 'react';
import PropTypes from 'prop-types';
import { Modal } from 'react-materialize';
import InputField from '../common/InputField';
import parseDate from '../../utilities/parseDate';

/**
* Role Card for viewing a single user
*
* @param {Object} props { title, id, description,
*  updatedAt, onChange, onSubmit, deleteRole }
* @returns {String} HTML markup for the Role Card
*/
const RoleCard = ({
    title, id, description, updatedAt, onChange, onSubmit, deleteRole }) => (
    <div>
      <h5>{title}</h5>
      <p>{description}</p>
      <p>Last Edited: {parseDate(updatedAt)}</p>
      <p />
      <p />
      <div className="col l6 m6 s12 left">
        <Modal
          trigger={
            <a
              className="waves-effect btn button-design"
              data-target="passwordModal" id="delete-role">
              Delete
                </a>}
          actions={
            <div>
              <button
                className="btn btn-flat modal-action modal-close"
                name={id} onClick={deleteRole}
                id="delete">
                Delete
                </button>
              <button
                className="btn btn-flat modal-action modal-close left">
                Cancel
                  </button>
            </div>}>
          <div >
            <h5
              className="center">
              Are you sure you want to delete the user
                </h5>
          </div>
        </Modal>
      </div>
      <div className="col l6 m6 s12 right">
        <Modal
          trigger={
            <a
              className="waves-effect btn button-design"
              data-target="passwordModal"
              id="update-role">
              Update
          </a>
          }
          actions={
            <div>
              <button
                className="btn btn-flat modal-action modal-close"
                type="submit"
                onClick={onSubmit}
                name={id}
                id="update">
                Update
                </button>
              <button
                className="btn btn-flat modal-action modal-close left">
                Cancel
                  </button>
            </div>}>
          <div>
            <h5>Update Role</h5>
            <InputField
              name="title"
              placeholder={title}
              className="validate form-design"
              type="text"
              onChange={onChange} />
            <InputField
              name="description"
              placeholder={description}
              className="validate form-design"
              type="text"
              onChange={onChange} />
          </div>
        </Modal>
      </div>
    </div>
  );

// Set Props
RoleCard.propTypes = {
  title: PropTypes.string,
  description: PropTypes.string,
  updatedAt: PropTypes.string,
  onChange: PropTypes.func,
  onSubmit: PropTypes.func,
  deleteRole: PropTypes.func,
  id: PropTypes.number
};

export default RoleCard;
