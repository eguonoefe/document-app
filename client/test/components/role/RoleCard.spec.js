/* global expect jest test */
import React from 'react';
import { shallow, configure } from 'enzyme';
import toJson from 'enzyme-to-json';
import Adapter from 'enzyme-adapter-react-16';
import RoleCard from '../../../components/role/RoleCard';

configure({ adapter: new Adapter() });

jest.dontMock('../../../components/role/RoleCard');

describe('RoleCard Component', () => {
  test('should match the role card snapshot', () => {
    const onChange = jest.fn();
    const onSubmit = jest.fn();
    const deleteRole = jest.fn();
    const component = shallow(
      <RoleCard
        title={'Book'}
        id={3}
        description={'Awesome'}
        onChange={onChange}
        onSubmit={onSubmit}
        deleteRole={deleteRole}
        createdAt={'today'} />);
    const tree = toJson(component);
    expect(tree).toMatchSnapshot();
  });
});
