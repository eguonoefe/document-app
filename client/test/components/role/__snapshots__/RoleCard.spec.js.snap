// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoleCard Component should match the role card snapshot 1`] = `
<div>
  <h5>
    Book
  </h5>
  <p>
    Awesome
  </p>
  <p>
    Last Edited: 
    Invalid Date
  </p>
  <p />
  <p />
  <div
    className="col l6 m6 s12 left"
  >
    <Modal
      actions={
        <div>
          <button
            className="btn btn-flat modal-action modal-close"
            id="delete"
            name={3}
            onClick={[MockFunction]}
          >
            Delete
          </button>
          <button
            className="btn btn-flat modal-action modal-close left"
          >
            Cancel
          </button>
        </div>
      }
      bottomSheet={false}
      fixedFooter={false}
      id="modal-1"
      options={
        Object {
          "dismissible": true,
          "endingTop": "10%",
          "inDuration": 250,
          "onCloseEnd": null,
          "onCloseStart": null,
          "onOpenEnd": null,
          "onOpenStart": null,
          "opacity": 0.5,
          "outDuration": 250,
          "preventScrolling": true,
          "startingTop": "4%",
        }
      }
      trigger={
        <a
          className="waves-effect btn button-design"
          data-target="passwordModal"
          id="delete-role"
        >
          Delete
        </a>
      }
    >
      <div>
        <h5
          className="center"
        >
          Are you sure you want to delete the user
        </h5>
      </div>
    </Modal>
  </div>
  <div
    className="col l6 m6 s12 right"
  >
    <Modal
      actions={
        <div>
          <button
            className="btn btn-flat modal-action modal-close"
            id="update"
            name={3}
            onClick={[MockFunction]}
            type="submit"
          >
            Update
          </button>
          <button
            className="btn btn-flat modal-action modal-close left"
          >
            Cancel
          </button>
        </div>
      }
      bottomSheet={false}
      fixedFooter={false}
      id="modal-1"
      options={
        Object {
          "dismissible": true,
          "endingTop": "10%",
          "inDuration": 250,
          "onCloseEnd": null,
          "onCloseStart": null,
          "onOpenEnd": null,
          "onOpenStart": null,
          "opacity": 0.5,
          "outDuration": 250,
          "preventScrolling": true,
          "startingTop": "4%",
        }
      }
      trigger={
        <a
          className="waves-effect btn button-design"
          data-target="passwordModal"
          id="update-role"
        >
          Update
        </a>
      }
    >
      <div>
        <h5>
          Update Role
        </h5>
        <InputField
          className="validate form-design"
          name="title"
          onChange={[MockFunction]}
          placeholder="Book"
          type="text"
        />
        <InputField
          className="validate form-design"
          name="description"
          onChange={[MockFunction]}
          placeholder="Awesome"
          type="text"
        />
      </div>
    </Modal>
  </div>
</div>
`;
